### ISSUE TITLE
Make fish look for mdocml?

### ISSUE THREAD

**Issue by Vintodrimmer** [fish-shell/fish-shell#3925]
Good day.

This [issue](https://github.com/voidlinux/void-packages/issues/6036#issuecomment-28929579) pretty much describes it.

If I understand it correctly, when the distribution uses  `mdocml` instead of 'man-db', then `fish_update_completions` fails with the following traceback:

```
Traceback (most recent call last):
  File "/usr/share/fish/tools/create_manpage_completions.py", line 929, in <module>
    file_paths.extend(get_paths_from_manpath())
  File "/usr/share/fish/tools/create_manpage_completions.py", line 860, in get_paths_from_manpath
    proc = subprocess.Popen(['manpath'], stdout=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 390, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1024, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
```

Is it reasonable to ask you to make fish look for `mdocml` if the `man-db` is not found or is it intended to work like that by design?

Fish version and terminal info:

```
fish --version
fish, version 2.5.0

echo $version
2.5.0

uname -a
Linux chrysalis 4.10.6-chrysalis #1 SMP PREEMPT Mon Mar 27 13:59:34 CEST 2017 x86_64 GNU/Linux

echo $TERM
rxvt-unicode-256color
```

**Comment by faho** [comment]
Already fixed in git, originally reported as #2194.

Note that, if you need a different $MANPATH than /usr/share/man:/usr/local/share/man, you'll need to define it yourself, since AFAIK mdoc doesn't support querying it (i.e. an analogue to `man --path`).

**Comment by faho** [comment]
And also:

>I've had some problems with very slow completions and decided to run fish_update_completions

Completions generated from man pages are barebones (they only do short and long options) and only used as fallback. So whatever was slow for you probably wasn't a man-generated completion so this wouldn't help.

Which completion was it?

**Comment by Vintodrimmer** [comment]
Ok, thank you and sorry for the duplicate. Search by  `mdocml` did not return anything.

I will wait for 2.6.0.

The completion in question had to do with `make` and linux kernel. I compile it with a custom config.

If I recall correctly, I typed in `make me` and pressed 'Tab', after which it just froze. I tried pressind 'right' and "up", but it didn't react at all. It ended either with 'ctrl+c' or on its own, I'm not quite sure.

I expected it to complete it to `make menuconfig`.

This happened at least twice with something simillar, so I asked on github.
